File: day05/getKeys.js
function getKeysFromValue(data, value) {
  var keys = [];
  for (var d in data) {
    if (data[d] == value) {
      keys.push(d);
    }
  }
    
  return value + ' is paired with: ' + keys;
}

function getData() {
  var dictionary = prompt('Please enter key/value pairs (e.g., key1: value1, key2: value2): ');
  var value = prompt('Enter a value to see how frequently it occurs: ');

  try {
    var data = {};
    var key_val = dictionary.split(', ');

    for (var k=0; k < key_val.length; k++) {
      var split = key_val[k].split(': ');
      data[split[0]] = split[1];
    }

    alert(getKeysFromValue(data, value));
  } catch(error) {
    console.log('Error:', error);
  }
}

File: day05/get_keys.py
def get_keys_from_value(data, value):
    keys = []
    for d in data:
        if data[d] == value:
            keys += [d]
    
    return str(value) + ' is paired with: ' + str(keys)

def get_data():
    dictionary = input('Please enter key/value pairs (e.g., key1: value1, key2: value2): ')
    value = input('Enter a value to see which keys it is paired with: ')

    try:
        data = {}
        key_val = dictionary.split(', ')

        for k in range(len(key_val)):
            split = key_val[k].split(': ')
            data[split[0]] = split[1]

        print(get_keys_from_value(data, value))
    except Exception as error:
        print('Error:', error)
        get_data()
    
if __name__ == '__main__':
    get_data()
